docs/release-cycle.md

```diff
@@ -0,0 +1,81 @@
+### kubeadm's release cycle and quarterly agenda
+
+Kubernetes has a well defined release process that improves with every release.
+kubeadm is a core component of Kubernetes.
+These notes are collected during the v1.8 release cycle, and provide some inspiration
+for how the Kubernetes developing flow actually can look like.
+
+Most of these tips and tricks are general and not kubeadm specific.
+But as we move more and more components out of the main Kubernetes repo, these communication
+channels and escalation paths become increasingly important. Some of them are explained
+in this doc.
+
+Some of what's outlined below are just rough guidelines and processes that we use when developing kubeadm.
+It makes sense to document this so the rest of the community can see what it looks like to be involved
+in developing a core component, and/or to onboard new contributors to the process.
+
+That said, nothing here is set in stone. See this as inspiration :)
+
+#### First month of the quarter (coding)
+
+ - Manage cherrypicks for the current stable branch
+   - Make sure high-priority bug reports are triaged and fixed.
+   - Create bug fixes, ensure they're merged, and cherry pick them to the release branch.
+   - Coordinate with the stable branch manager.
+ - Cleanup work. As kubeadm supports deploying clusters one minor version below the CLI version, support for v1.7-specific workarounds may be dropped when the v1.9 cycle starts
```

Comment:
Doing this too early could make it difficult to cherry pick bug fixes back to the release branch (especially if it involves large refactors).

Reply:
Indeed. Should I point that out here?